### PR TITLE
update: tep-0042 user stories and alternatives

### PR DESCRIPTION
**Pull request by waveywaves**
Add user story surrounding CLI usage and the considered
alternatives for the API format including the unused formats and some
of their pitfalls.

Created as a followup of https://github.com/tektoncd/community/pull/308#issuecomment-802056087
* [x]  The proposed syntax is still using `onFailure: true` vs a list syntax or an "enum" syntax
* [x]  I think [your comments about the different directions we want to go](https://github.com/tektoncd/community/pull/308#discussion_r592756261) would be great to include in the TEP itself (e.g. in notes or alteratives maybe?)
* [x]  It would be great to describe something like [this conversation](https://github.com/tektoncd/community/pull/308#discussion_r595520939) around how a user and/or integrating tool would interact with this functionality

cc @bobcatfish @vdemeester

### CODE REVIEW

File: teps/0042-taskrun-breakpoint-on-failure.md [hunk 1]
```diff
@@ -122,3 +133,75 @@ failed step 0 as a success. Running this script would create `/tekton/tools/0.er
...
+   ```yaml
+   metadata:
+     name: failed-taskrun-1234-debug
+   spec:
+    debug: 
+      breakpoint: ["onFailure"]
+   ```
+3. Once the TaskRun is created and a step fails, the TaskRun Pod will be in the Running state. Till that time the CLI will wait 
+   for the TaskRun to stop executing (due to failure) and go into the limbo state which would be leveraged for debugging.
+   The CLI will open a shell to the step container which would be a reimplementation of `kubectl exec -it`.
```
Comment (bobcatfish) [inline teps/0042-taskrun-breakpoint-on-failure.md:158]: 🎉  nice!! thanks for explaining this

File: teps/0042-taskrun-breakpoint-on-failure.md [hunk 2]
```diff
@@ -38,8 +49,8 @@ kind: TaskRun
 metadata:
   generateName: quarante-deux
 spec:
-  breakpoint:
-    onFailure: true
+  debug:
+    breakpoint: ["onFailure"]
```
Comment (bobcatfish) [inline teps/0042-taskrun-breakpoint-on-failure.md:53]: wooot

File: teps/0042-taskrun-breakpoint-on-failure.md [hunk 3]
```diff
@@ -122,3 +133,75 @@ failed step 0 as a success. Running this script would create `/tekton/tools/0.er
...
+    debug:
+      breakpoint: 
+        onFailure: 
+          before: true
+          after: true
+        beforeStep: []
+        afterStep: []
+    ```
+
+3. *Enums rock*
```
Comment (bobcatfish) [inline teps/0042-taskrun-breakpoint-on-failure.md:189]: 😆

**Review by bobcatfish — COMMENTED** [review]
looks great to me @waveywaves !!

/approve

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**